arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/Schedule.kt

```diff
@@ -192,21 +212,48 @@ sealed class Schedule<Input, Output> {
   /**
    * Conditionally checks on both the input and the output whether or not to continue.
    */
-  abstract fun <A : Input> check(pred: suspend (A, Output) -> Boolean): Schedule<A, Output>
+  abstract fun <A : Input> check(pred: suspend (input: A, output: Output) -> Boolean): Schedule<A, Output>
 
   /**
    * Inverts the decision of a schedule.
    */
   abstract operator fun not(): Schedule<Input, Output>
 
-  /**
-   * Combines with another schedule by combining the result and the delay of the [Decision] with the functions [f] and [g]
-   */
-  abstract fun <A : Input, B> combineWith(
+  @Deprecated(
+    "combineWith is deprecated in favor of combineNanos and combine using Double/kotlin.time.Duration instead of arrow.fx.coroutines.Duration.",
```

Comment:
wouldn't be easier replacing with the kotlin.Duration version?

Reply:
Yes it absolutely would but `kotlin.time.Duration` is still `@Experimental` and this would make a large portion of the `Schedule` API `@Experimental` as well while in it's not experimental at all.

Reply:
Makes sense, but I still believe that replacing Duration with nanos is not a good idea. I'd personally rather do a migration to something experimental but do it once than do it to something flaky (like nanos with Double) to then migrate again to Duration.

In summary: 

* Promoting nanos API -> we're promoting a flaky API (I believe it's really easy switching millis with nanos and break something) + users will likely need to do an additional migration to Duration
* Promoting Duration API -> we promote what's in Kotlin + there's no extra migration needed + users can still choose the nanos API if they don't want to deal with experimental APIs

Reply:
>  I still believe that replacing Duration with nanos is not a good idea

We need to use `Double` nanos in the implementation or the implemenation itself becomes `@ExperimentalTime` which means that *all* APIs would become `@ExperimentalTime` also when they don't rely on `Duration` in their API anywhere.

I don't want to promote `Double` as nanos in the public APIs, and I'd rather also not expose it but at this point we don't really have a choose. We could also use `Long` and `milliseconds` in the public API, I guess that is more common and similar to what KotlinX Coroutines does with `delay`.

Reply:
Not to mention that we also need a notion of infinity https://github.com/arrow-kt/arrow-fx/issues/18